tgui/packages/tgui/interfaces/GasCanister.js

```diff
@@ -0,0 +1,110 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../backend';
+import { Divider, Box } from '../components';
+import { Window } from '../layouts';
+import { PortableBasicInfo, PortableHoldingTank } from './common/PortableAtmos';
+import { ReleaseValve } from './common/ReleaseValve';
+import { Detonator } from './GasCanister/Detonator';
+
+export const GasCanister = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const {
+    connected,
+    holding,
+    hasValve,
+    valveIsOpen,
+    pressure,
+    maxPressure,
+    releasePressure,
+    minRelease,
+    maxRelease,
+    detonator,
+    detonatorAttachments,
+  } = data;
+
+  const handleSetPressure = releasePressure => {
+    act("set-pressure", {
+      releasePressure,
+    });
+  };
+
+  const handleToggleValve = () => {
+    act("toggle-valve");
+  };
+
+  const handleEjectTank = () => {
+    act("eject-tank");
+  };
+
+  const handleWireInteract = (toolAction, index) => {
+    act("wire-interact", {
+      index,
+      toolAction,
+    });
+  };
+
+  const handleToggleAnchor = () => {
+    act("anchor");
+  };
+
+  const handleToggleSafety = () => {
+    act("safety");
+  };
+
+  const handlePrimeDetonator = () => {
+    act("prime");
+  };
+
+  const handleTriggerActivate = () => {
+    act("trigger");
+  };
+
+  const handleSetTimer = newTime => {
+    act("timer", {
+      newTime,
+    });
+  };
+
+  let hasDetonator = !!(detonator);
+
+  return (
+    <Window
+      resizable
+      key={hasDetonator}
```

Comment:
Shouldn't need a `key` prop as this isn't rendered as part of an array of things.

Reply:
I think I ran into an error before if I didn't have it but uuuh let's remove it and see what happens